### ISSUE TITLE
--disallow-any-generics generates spurious messages

### ISSUE THREAD

**Issue by msullivan** [python/mypy#7077]
In certain cases, `--disallow-any-generics` will generate an error message in the file where a type alias is used with the line number at which the type alias is defined.

For example
```
[case testDisallowAnyGenericsMessages]
# mypy: disallow-any-generics
from a import B
x: B

[file a.py]
from typing import TypeVar, List

T = TypeVar('T')

A = List[T]
B = A

[builtins fixtures/list.pyi]
```
generates the error `main:6: error: Missing type parameters for generic type`. `main:6` is not a line that appears.

The error here is line 6 of `a.py`, `B = A`, though that is not an error in this case because `disallow-any-generics` is not specified for that file. It is triggered by line 3 of main, but reporting an error *there* would also be wrong because there is nothing wrong with line 3 of main, just with the definition of `B`. (And there is nothing that can be fixed at that call site, because `B` is not generic!).

I'm marking this as high priority because `disallow-any-generics` is a super important option that we I think we should make a default once it is fixed.

**Comment by ilevkivskyi** [comment]
In general it seems to me we should prohibit using types as contexts for errors in general (or make file name part of error context).

**Comment by JukkaL** [comment]
I think that types are probably fine as error contexts in the semantic analyzer.

**Comment by happy2mylife** [comment]
Sorry, I have problem.

```
from typing import TypeVar, Generic

T = TypeVar('T', bound='Test')
class Test(Generic[T]):

    def __str__(self) -> str:
        return ""
```

test.py:3: error: Missing type parameters for generic type

I can't understand this error.

**Comment by res0nanz** [comment]
I also have same problem with @happy2mylife .

**Comment by happy2mylife** [comment]
Is this problem not solved yet? @msullivan

**Comment by JelleZijlstra** [comment]
You probably need to put `Test[Any]` in your bound. This example looks rather different from the original one in this bug.

**Comment by res0nanz** [comment]
If do the following,  cause `[flake8] F401:'typing.Any' imported but unused`.
```python
from typing import Any
T = TypeVar('T', bound='Test[Any]')
```
But if not import, `[mypy] error:Name 'Any' is not defined`.  How can I solve it?

I'm sorry in clumsy sentence.
